client/landing/gutenboarding/stores/onboard/types.ts

```diff
@@ -38,4 +38,7 @@ export interface Design {
 	fonts: FontPair;
 	categories: Array< string >;
 	is_premium: boolean;
+	site_options?: {
+		[ option: string ]: string | number | boolean;
+	};
```

Comment:
I think I'd prefer if we'd allowed a more limited range of attributes that are more descriptive of a design here. E.g. IMO it makes sense to say that a design requires the site editor, or "is an" FSE theme, but I'd start by just adding an option for just that.

Some of this is, once more, because of YAGNI -- we currently need the site editor and Gutenberg edge stickers, and AFAICS, we always need them both -- so I'd go with a single (possibly boolean) option to state that a design needs them. Note that @ramonjd is adding `is_edge` attributes in #42354 -- that might overlap with what we need?

The other reason is that I think it makes sense to keep the API surface for designs small, and descriptive of the design (and aligns better with the mental model). A design obviously has a title, comes with a font pairing, can be premium or not -- so these things are fairly descriptive. OTOH, the `site_options` it needs to work aren't really a core property of a design.

Reply:
The reason I like this approach is so easy to use and extend. It's maybe 10loc to support any case we need. Additionally, this would allow us to override any of the site creation options during the site creation phase on a per-design level, which I can see being very useful. 🤔 

Though I don't disagree with YAGNI in general, I feel like that approach would be more brittle in this case. For example, in order to set the gutenberg edge sticker on a per-design basis, we'd have to do the following:

1. Add `is_gutenberg_edge` to design 
2. Pass that through site creation options
3. Access it in the API and if true, apply the gutenberg edge sticker.

We'd have to do a similar thing for the site editor. And for any other item we might want to add.

I guess, do you think YAGNI still applies even when coding the more specific approach is a similar effort? I.e. the general approach here isn't really any extra effort, and it has the benefit of being useful for other things in the future. In fact, the general approach would be easier to maintain, in my opinion.

But I do appreciate you bringing this up. I'm not 100% sure about it either and wanted to start a discussion :)

> `is_edge` attributes

I think this just indicates which environment a design can show up in? e.g. a design shows up on horizon/wpcalypso/localhost, but not prod

Reply:
> I think this just indicates which environment a design can show up in? e.g. a design shows up on horizon/wpcalypso/localhost, but not prod

That's right. I chose `is_edge` just to mean _"a new design we want to test but not release quite yet"_ without reflecting much on terminology. 

We can change it if folks have an alternative that makes more sense. The PR isn't merged yet.

Reply:
@ockham I've developed https://github.com/Automattic/wp-calypso/pull/43349 a bit further, and I think we can avoid having options & stickers there :)